### ISSUE TITLE
[Bug]: Item Blacklist does not seem to work

### ISSUE THREAD

**Issue by JetString** [promcteam/proskillapi#205]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

ProSkillAPI-1.1.7.11 - 1.1.7.13
I am currently using 1.1.7.11 because of another issue which i will report separately.

When restricting certain classes to certain armor items. The classes can still use all items. They are not restricted at all.

### Expected Behavior

The classes should not be able to use blacklisted items.

### Steps To Reproduce

  blacklist: 
  - 'IRON_CHESTPLATE'
  - 'DIAMOND_BOOTS'
  - 'NETHERITE_BOOTS'
  - 'DIAMOND_LEGGINGS'
  - 'IRON_LEGGINGS'
  - 'NETHERITE_HELMET'
  - 'NETHERITE_LEGGINGS'
  - 'NETHERITE_CHESTPLATE'
  - 'IRON_HELMET'
  - 'DIAMOND_HELMET'
  - 'DIAMOND_CHESTPLATE'
  - 'IRON_BOOTS'
  
  This should work. But it doesn't.

### Environment

```markdown
- Server Software: paper mc 1.18.1 build 133 
- ProSkillAPI Version: 1.1.7.11 - 1.1.7.13
- ProMCCore Version: promccore-1.0.3.5-ALPHA
- Plugins:
advanced portals-0.8.0
antiRedstone Clock-1.5.0
Autorank-4.5.1
Boss-4.1.7
Citizens-2.0.29-b2420
coloredSigns-5.0.0
Combat WeaponryPlus-1.4.7
DiscordSRV-Build-1.24.0
dtlTraders-6.4.7
Dungeons
Elevator
EliteArmor-Recode-6.5.1
EssentialsX-2.19.3-dev+6-739600e
EssentialsXAntiBuild-2.19.3-dev+6-739600e
EssentialsXChat-2.19.3-dev+6-739600e
EssentialsXDiscord-2.19.3-dev+6-739600e
EssentialsXProtect-2.19.3-dev+6-739600e
EssentialsXSpawn-2.19.3-dev+6-739600e
ExcellentEnchants-3.0.8
GPS
GPSQuests-1.5
Graves-4.5
GriefPrevention
LevelledMobs-3.3.2 b596
LocaleLib-2.4
LuckPerms-Bukkit-5.3.86
LWCX-2.2.7
Magic-10.5.1
mcMMO
minimotd-universal-2.0.5
MobArena
MoneyFromMobs
Multiverse-Core-4.3.2-SNAPSHOT
Multiverse-Portals-4.2.2-SNAPSHOT
MyPet-3.12-SNAPSHOT-B1667
NexEngine
nuvotifier
PlaceholderAPI-2.11.1
promccore-1.0.3.5-ALPHA
ProSkillAPI-1.1.7.11-ALPHA
ProtocolLib
pvparena-1.15.1
Quests-4.1.3
UltimateAutoRestart+Build+48a
Vault
VotingPlugin
WorldBorder
worldedit-bukkit-7.2.9-SNAPSHOT-dist
WorldEditSUI-1.7.2
worldguard-bukkit-7.0.7-SNAPSHOT-dist
```

### latest.log

no error logs.

### Anything else?

Yo, I finally listed all my plugins.
:)

**Comment by goflishMC** [comment]
@JetString next time you can just do /pl and it will give you a list and you can screenshot it.

**Comment by lordsrubka5** [comment]
I mentioned this on discord yesterday I think. From my observation skillapi does not block you fully from using those items, however if the item is blacklisted you will not gain skillapi attributes from the said item. This is very useful for me since my staff is a hoe and I still want people to use hoes for farming. If you use prorpgitems you will have to add the class restrictions to the items to prevent rpgitem stats from being given.

**Comment by JetString** [comment]
> @JetString next time you can just do /pl and it will give you a list and you can screenshot it.

(-_Q)

**Comment by Travja** [comment]
There's two ways to interpret this, one would be that when an item is blacklisted, you can't use it at all (no equipping, not holding, nothing). The other would be that you can still use the item but get no benefit from it (as it is right now).

**Comment by Travja** [comment]
Looking further at the code for the blacklist, it looks like either the admin can set the items to be dropped if they're unusable, or if they don't, the item can still be held and interacted with. But damage done by the item is cancelled, damage blocked by shields that are blacklisted should be cancelled. If bows are blacklisted, they shouldn't be able to shoot.

Now, I can still add an option if you'd like to full-block the item, essentially making it an obtainable only item which can't be equipped or held in your hand if that's the better option. I may do that anyway just for configurability's sake.

**Comment by JetString** [comment]
the more options the better i guess.
well particularly for armor items the latter is the better option i think.
